cache/cache.go

```diff
@@ -120,11 +136,19 @@ func (c *Cache) Update(u interface{}, value interface{}) (err error) {
 
 // AddOrUpdate adds a new value in the cache or updates the existing value
 // if needed. If an update happens the timestamp is also updated.
-func (c *Cache) AddOrUpdate(u interface{}, value interface{}) (err error) {
+func (c *Cache) AddOrUpdate(u interface{}, value interface{}) {
 
 	var timer *time.Timer
 	if c.lifetime != -1 {
-		timer = time.AfterFunc(c.lifetime, func() { c.Remove(u) })
+		timer = time.AfterFunc(c.lifetime, func() {
+			if err := c.Remove(u); err != nil {
+				log.WithFields(log.Fields{
```

Comment:
This has caught lots of other issues. Its a safety valve. Thats why I am leaving it there.